packages/state-transition/src/util/seed.ts

```diff
@@ -135,6 +138,46 @@ export function computeProposerIndex(
   );
 }
 
+/**
+ * Return the proposer indices for the given `epoch`.
+ * A more generic version of `computeProposers`
+ */
+export function computeProposerIndices(
+  fork: ForkSeq,
+  state: CachedBeaconStateAllForks,
+  epoch: Epoch
+): ValidatorIndex[] {
+  const startSlot = computeStartSlotAtEpoch(epoch);
+  const proposers = [];
+  const epochSeed = getSeed(state, epoch, DOMAIN_BEACON_PROPOSER);
+  // TODO FULU: Compute shuffling if shuffling cache miss is a temporary workaround.
+  // EpochCache needs to cache shuffling for nextEpoch + 1 too.
+  let shuffling: EpochShuffling;
+  try {
+    shuffling = state.epochCtx.getShufflingAtEpoch(epoch);
```

Comment:
this is not correct because shuffling of `epoch + 2` is not available at this time so this line will always fail
we should do the query from `shufflingCache` similar to https://github.com/ChainSafe/lodestar/blob/v1.31.0/packages/state-transition/src/cache/epochCache.ts#L656

need to add a metric of number of times of `hit/miss` or see if it's available in the shuffling cache
hopefully most of the time shuffling of `epoch + 2` is ready